sw/device/tests/sim_dv/csrng_lc_hw_debug_en_test.c

```diff
@@ -0,0 +1,280 @@
+}
+
+static void fw_override_conditioner_write(
+    const dif_entropy_src_t *entropy_src) {
+  CHECK_DIF_OK(dif_entropy_src_conditioner_start(entropy_src));
+
+  const uint32_t kInputMsg[kEntropyFifoBufferSize] = {
+      0xa52a0da9, 0xcae141b2, 0x6d5bab9d, 0x2c3e5cc0, 0x225afc93, 0x5d31a610,
+      0x91b7f960, 0x0d566bb3, 0xef35e170, 0x94ba7d8e, 0x534eb741, 0x6b60b0da,
+  };
+
+  uint32_t fail_count = 0;
+  uint32_t total = 0;
+  do {
+    uint32_t count;
+    dif_result_t op_result = dif_entropy_src_observe_fifo_write(
+        entropy_src, kInputMsg + total, ARRAYSIZE(kInputMsg) - total, &count);
+    total += count;
+    if (op_result == kDifIpFifoFull) {
+      CHECK(fail_count++ < kTestParamEntropySrcMaxAttempts);
+    } else {
+      fail_count = 0;
+      CHECK_DIF_OK(op_result);
+    }
+  } while (total < ARRAYSIZE(kInputMsg));
+  entropy_conditioner_stop(entropy_src);
+}
+
+static void csrng_static_generate_run(uint32_t *output, size_t output_len) {
+  entropy_testutils_stop_all();
+  // TODO: May need to flush the output buffers before enabling enabling
+  // firmware override connected to csrng.
+  entropy_testutils_fw_override_enable(&entropy_src, kEntropyFifoBufferSize,
+                                       /*firmware_override_enable=*/false,
+                                       /*bypass_conditioner=*/false);
+  CHECK_DIF_OK(dif_csrng_configure(&csrng));
+  fw_override_conditioner_write(&entropy_src);
+
+  const dif_csrng_seed_material_t kEmptySeedMaterial = {0};
+  CHECK_DIF_OK(dif_csrng_instantiate(&csrng, kDifCsrngEntropySrcToggleEnable,
+                                     &kEmptySeedMaterial));
+
+  csrng_testutils_cmd_generate_run(&csrng, output, output_len);
+  uint32_t prev_word = 0;
+  for (size_t i = 0; i < output_len; ++i) {
+    CHECK(prev_word != output[i],
+          "Unexpected duplicate value at index: %d value: 0x%x", i, prev_word);
+    prev_word = output[i];
+  }
+}
+
+bool test_main(void) {
+  init_peripherals();
+  flash_ctrl_testutils_default_region_access(&flash_ctrl_state,
+                                             /*rd_en=*/true,
+                                             /*prog_en=*/true,
+                                             /*erase_en=*/true,
+                                             /*scramble_en=*/false,
+                                             /*ecc_en=*/false,
+                                             /*he_en=*/false);
+
+  dif_rstmgr_reset_info_bitfield_t rst_info = rstmgr_testutils_reason_get();
+  rstmgr_testutils_reason_clear();
+
+  dif_lc_ctrl_state_t lc_state;
+  CHECK_DIF_OK(dif_lc_ctrl_get_state(&lc_ctrl, &lc_state));
+  LOG_INFO("lc state: %d", lc_state);
+
+  if (rst_info == kDifRstmgrResetInfoPor &&
+      lc_state == kDifLcCtrlStateTestUnlocked0) {
+    enum {
+      kPartitionId = 0,
+    };
+    uint32_t address =
+        (uint32_t)(nv_csrng_output)-TOP_EARLGREY_FLASH_CTRL_MEM_BASE_ADDR;
+    if (kDeviceType == kDeviceSimVerilator) {
```

Comment:
I THINK this erase might not be needed anymore.
@a-will added verilator initialization such that the flash looks erased.

Reply:
Come to think of it... I should've erased the entire flash, not just the first bank. It'll just have to have a different name for now.

Reply:
Removed.